### PR TITLE
fix: federation banner having no text after adding a federated user FS-72

### PR DESCRIPTION
**Pull request by typfel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-72" title="FS-72" target="_blank">BugFS-72</a>  [iOS] Distinguish "federated users are present" from guests
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Federation banner has no text after adding a federated user.

### Causes (Optional)

The banner view is updated before the view is loaded and it has a condition to ignore updates before the view is loaded.

### Solutions

Store state if the view is not loaded and apply it after the view is loaded.

### Testing

- create a group
- don't add any federated participant - only users from same backend
- go inside the group details
- add federated users
- look for the banner - banner doesn't show the text

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift [hunk 1]
```diff
@@ -44,7 +44,6 @@ final class GuestsBarController: UIViewController {
             return _state
         }
         set {
-            guard newValue != state else { return }
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift:47]: This guard existed twice, now it only exists in `setState`

File: Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift [hunk 2]
```diff
@@ -81,24 +81,30 @@ final class GuestsBarController: UIViewController {
     // MARK: - State Changes
 
     func setState(_ state: State, animated: Bool) {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift:83]: I pulled out update logic from setState(state:animated) into a separate private method `updateState(animated:)`

File: Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift [hunk 3]
```diff
@@ -81,24 +81,30 @@ final class GuestsBarController: UIViewController {
...
+            return
+        }
+
         configureTitle(with: state)
         let collapsed = state == .hidden
 
         let change = {
             if !collapsed {
                 self.heightConstraint.constant = collapsed ? GuestsBarController.collapsedHeight : GuestsBarController.expandedHeight
-                self.view.setNeedsLayout()
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift:93]: These caused issues animation issues after I updated the state in `viewDidLoad`. Did some testing and everything behaves normally without them.

File: Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift [hunk 4]
```diff
@@ -119,10 +125,10 @@ final class GuestsBarController: UIViewController {
         case .hidden:
             label.text = nil
             label.accessibilityIdentifier = nil
-        case .visible(let labelKey, let accessibilityIdentifier):
-            let markdownTitle = labelKey.localized
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/GuestBar/GuestsBarController.swift:123]: The text is already localized at this point.

**Review by sb88k — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**